### PR TITLE
Ingredient APIs v0.1

### PR DESCRIPTION
**Pull request by BXLnightManu**

### CODE REVIEW

File: src/handlers/ingredient.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            ok: false,
+            payload: 'Must provide an object like: {name: "string", newValue: "string"}'
+        });
+    } else {
+        const name = req.body.name;
+        const newValue = req.body.newValue;
+        Ingredient.findOne({name})
+            .then( doc => {
+                if(doc) {
+                    return Ingredient.findOneAndUpdate( {_id:`${doc._id}`}, {$set: {name: `${newValue}`}}, {new: true} )
```
Comment (ngasst) [inline src/handlers/ingredient.js]: Do not needlessly interpolate variables? That means, no need to use `${}` unless needed!

File: src/handlers/ingredient.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                } else {
+                    return false;
+                }
+            })
+            .then(newDoc => {
+                if(!newDoc) {
+                    res.json({ ok: false, payload: `The ingredient "${name}" does not exist ! Pleas provide an existing name of ingredient.`})
```
Comment (ngasst) [inline src/handlers/ingredient.js]: Typo. Did you mean "Please"?

File: src/handlers/ingredient.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                }
+            })
+            .catch(err => {
+                console.log(err);
+                res.json({ ok: false, payload: err.message || "FAILED" });
+            });
+    }
+}
+
+function erise(req, res) {
```
Comment (ngasst) [inline src/handlers/ingredient.js]: typo. Did you mean "erase"?

File: src/handlers/ingredient.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    if(!req.body.name || typeof req.body.name !=="string") {
```
Comment (ngasst) [inline src/handlers/ingredient.js]: Though HTTP spec does indeed allow to pass a payload along with your DELETE request, the convention and common practice is to only provide a URL param that uniquely identifies the resource you need to delete. Please correct accordingly.

File: src/handlers/ingredient.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        res.json({ 
+            ok: false,
+            payload: 'Must provide an object like: {name: "string"}'
+        });
+    } else {
+        const name = req.body.name;
+        Ingredient.findOne({name})
```
Comment (ngasst) [inline src/handlers/ingredient.js]: There is no need to find only to find and delete. That's why the method is called `findAndRemove`.

File: src/handlers/ingredient.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        .then(doc => {
+            if(doc) {
+                Ingredient.findOneAndRemove( {_id:`${doc._id}`});
+                res.json({ ok: true, payload: `The ingredient "${name} has been successfully deleted.` });
+            } else {
+                res.json({ ok: false, payload: `The ingredient "${name}" does not exist ! Pleas provide an existing name of ingredient.` });
```
Comment (ngasst) [inline src/handlers/ingredient.js]: Typo. Did you mean "Please"?

File: src/routes/ingredient.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const express = require('express');
+const ingredientRouter = express.Router();
+const { ingredient } = require('../handlers');
+
+// define client routes and their handlers
+// LIST
+ingredientRouter.get('/', ingredient.list);
+ingredientRouter.post('/', ingredient.create);
+ingredientRouter.put('/', ingredient.update);
+ingredientRouter.delete('/', ingredient.erise);
```
Comment (ngasst) [inline src/routes/ingredient.js]: See above for 'erise' typo.

File: src/handlers/ingredient.js (hunk 1 above)
Comment (BXLnightManu) [reply src/handlers/ingredient.js]: Will be corrected in the next commit.

File: src/handlers/ingredient.js (hunk 2 above)
Comment (BXLnightManu) [reply src/handlers/ingredient.js]: Yes. Will be corrected in the next commit.

File: src/handlers/ingredient.js (hunk 3 above)
Comment (BXLnightManu) [reply src/handlers/ingredient.js]: I replaced it by remove.

File: src/handlers/ingredient.js (hunk 5 above)
Comment (BXLnightManu) [reply src/handlers/ingredient.js]: Will be corrected in the next commit.

File: src/routes/ingredient.js (hunk 7 above)
Comment (BXLnightManu) [reply src/routes/ingredient.js]: Will be corrected in the next commit.

**Pull request merged**